ArduCopter/FlightMode.h

```diff
@@ -1002,76 +1033,28 @@ class FlightMode_Throw : public FlightMode {
 };
 
 
-
-class FlightMode_Guided_NoGPS : public FlightMode_Guided {
-
-public:
-
-    FlightMode_Guided_NoGPS(Copter &copter) :
-        Copter::FlightMode_Guided(copter)        { }
-
-    bool init(bool ignore_checks) override;
-    void run() override;
-
-    bool requires_GPS() const override { return true; }
-    bool has_manual_throttle() const override { return false; }
-    bool allows_arming(bool from_gcs) const override {
-        if (from_gcs) {
-            return true;
-        }
-        return false;
-    }
-    bool is_autopilot() const override { return true; }
-
-protected:
-
-    const char *name() const override { return "GUIDED_NOGPS"; }
-    const char *name4() const override { return "GNGP"; }
-
-private:
-
-};
-
-
-class FlightMode_SmartRTL : public FlightMode_RTL {
+class FlightMode_Avoid_ADSB : public FlightMode_Guided {
```

Comment:
Thanks for the review, I had compile problems because _Avoid_ADSB is a subclass of Guided.  I tried just doing a simple declaration of Guided at the top but couldn't get it to work.  I'll consult with Peter about this and maybe try again.